### PR TITLE
Markup API doc in Markdown style

### PR DESCRIPTION
**Pull request by ybiquitous**
- Specify `--markup markdown` option to YARD.
- Add and publish `.yardopts` file, instead of giving the option in `Rakefile`.
  This `.yardopts` file allows to use the same options on <https://www.rubydoc.info/>.
- Rewrite the `Capybara.configure` method doc because it's broken in Markdown style. See below.
  - The latest doc is [here](https://www.rubydoc.info/gems/capybara/3.21.0/Capybara#configure-class_method).

[skip ci]

## `Capybara.configure`  change

In addition, I re-sort the options in alphabetical order because I think too hard to find an option in the long list.
But this big change may cause a discussion. Please give me any feedback! 

### Before

Run `bundle exec rake yard`:
image

### After

image

### CODE REVIEW

**Comment by ybiquitous** [comment]
@twalpole My documentation modification started from #2190 is last with this PR. I believe there are no other places which we should fix about documentation. Thanks a lot for your kindness. 😄

File: .gitignore [hunk 1]
```diff
@@ -21,3 +21,4 @@ bundler_stubs
 bin
 gem-private_key.pem
 save_path_tmp
+vendor/bundle
```
Comment (twalpole) [inline .gitignore]: Why was this necessary?

**Comment by twalpole** [comment]
I would prefer the configure doc changes to have the type and default on the same line as the option name if possible - I find it tough to scan for anything the way it is now.

File: .gitignore (hunk 1 above)
Comment (ybiquitous) [reply .gitignore]: Sorry, I miss the reason. 🙏 

I prefer a style to use the [`bundle install --deployment`](https://bundler.io/v2.0/man/bundle-install.1.html#DEPLOYMENT-MODE) command, which installs the gems to `vendor/bundle` directory.
If you don't like it, I'll remove it. 👍

**Comment by ybiquitous** [comment]
Make sense. I'll try another better solution. 💪 

BTW, when I rebased #2209, `Capybara::Selector` doc breaks, so I'll fix it also.

File: lib/capybara.rb [hunk 2]
```diff
@@ -66,39 +66,145 @@ class << self
     #       config.app_host   = 'http://www.google.com'
     #     end
     #
-    # === Configurable options
-    #
-    # [app_host = String/nil]             The default host to use when giving a relative URL to visit, must be a valid URL e.g. http://www.example.com
```
Comment (ybiquitous) [inline lib/capybara.rb]: [question] A description about `app_host` default value is missing, is it no problem?

**Comment by ybiquitous** [comment]
I changed from multi-line style to one-line style via 00376db3e088f6da9e41f2ee3d7dd3ed8298dd3a, how about?

```markdown
- **{name}** ({Type} = `{default value}`) - {description}

- **allow_gumbo** (Boolean = `false`) - When `nokogumbo` is available, whether it will be used to parse HTML strings.
```

Screenshot:

image

File: lib/capybara/selector.rb [hunk 3]
```diff
@@ -19,156 +19,156 @@
...
 #   * Locator: Match id or name
 #   * Filters:
-#     * :id (String, Regexp, XPath::Expression) - Match id attribute
-#     * :name (String) - Match name attribute
-#     * :class (String, Array<String>, Regexp, XPath::Expression) - Matches the class(es) provided
-#     * :style (String, Regexp, Hash)
+#       * :id (String, Regexp, XPath::Expression) - Match id attribute
+#       * :name (String) - Match name attribute
+#       * :class (String, Array<String>, Regexp, XPath::Expression) - Matches the class(es) provided
+#       * :style (String, Regexp, Hash)
```
Comment (ybiquitous) [inline lib/capybara/selector.rb]: [note] It's necessary to deepen 3rd level indentation, please see an example below (which was one before the update).

image

File: lib/capybara/selector.rb [hunk 4]
```diff
@@ -19,156 +19,156 @@
...
 # * **:checkbox** - Find checkboxes
 #   * Locator: Match id, {Capybara.configure test_id} attribute, name, or associated label text
 #   * Filters:
-#     * *:id (String, Regexp, XPath::Expression) - Matches the id attribute
-#     * *:name (String) - Matches the name attribute
-#     * *:class (String, Array<String>, Regexp, XPath::Expression) - Matches the class(es) provided
-#     * *:checked (Boolean) - Match checked fields?
-#     * *:unchecked (Boolean) - Match unchecked fields?
-#     * *:disabled (Boolean) - Match disabled field?
-#     * *:option (String) - Match the value
```
Comment (ybiquitous) [inline lib/capybara/selector.rb]: [note] I think they are with an extra asterisk(`*`).

**Comment by ybiquitous** [comment]
I fixed the 2 problems, please review again and give me feedbacks. 🙏

File: .gitignore (hunk 1 above)
Comment (twalpole) [reply .gitignore]: Ok -- it's fine

File: lib/capybara.rb (hunk 2 above)
Comment (twalpole) [reply lib/capybara.rb]: It defaults to nil - so it's fine

File: lib/capybara/selector.rb (hunk 4 above)
Comment (twalpole) [reply lib/capybara/selector.rb]: yes

**Comment by twalpole** [comment]
LGTM - Thanks a lot for all this work 👏

**Comment by ybiquitous** [comment]
You are welcome.
Thank you so much, too! ☺️
